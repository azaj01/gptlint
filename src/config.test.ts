import { expect, test } from 'vitest'

import { parseLinterConfig } from './config.js'

test('parseLinterConfig - valid', () => {
  expect(parseLinterConfig({})).toMatchSnapshot()
  expect(
    parseLinterConfig({
      ruleFiles: ['rules/**/*.md'],
      llmOptions: {
        model: 'gpt-4-turbo-preview',
        weakModel: 'gpt-4o-mini'
      }
    })
  ).toMatchSnapshot()

  expect(
    parseLinterConfig({
      files: ['src/**/*.{ts,tsx,js,jsx}'],
      llmOptions: {
        apiBaseUrl: 'https://openrouter.ai/api/v1',
        model: 'anthropic/claude-3-opus:beta',
        weakModel: 'anthropic/claude-3-haiku:beta',
        kyOptions: {
          headers: {
            'HTTP-Referer': 'https://gptlint.dev',
            'X-Title': 'gptlint'
          }
        }
      }
    })
  ).toMatchSnapshot()
})

test('parseLinterConfig - invalid', () => {
  expect(() => parseLinterConfig({ unknown: true } as any)).toThrow()
  expect(() =>
    parseLinterConfig({
      files: [],
      llmOptions: { model: false }
    } as any)
  ).toThrow()
})
