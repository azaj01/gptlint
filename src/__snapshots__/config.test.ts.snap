// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`parseLinterConfig - valid 1`] = `{}`;

exports[`parseLinterConfig - valid 2`] = `
{
  "llmOptions": {
    "model": "gpt-4-turbo-preview",
    "weakModel": "gpt-4o-mini",
  },
  "ruleFiles": [
    "rules/**/*.md",
  ],
}
`;

exports[`parseLinterConfig - valid 3`] = `
{
  "files": [
    "src/**/*.{ts,tsx,js,jsx}",
  ],
  "llmOptions": {
    "apiBaseUrl": "https://openrouter.ai/api/v1",
    "kyOptions": {
      "headers": {
        "HTTP-Referer": "https://gptlint.dev",
        "X-Title": "gptlint",
      },
    },
    "model": "anthropic/claude-3-opus:beta",
    "weakModel": "anthropic/claude-3-haiku:beta",
  },
}
`;
